src/test/java/com/puppycrawl/tools/checkstyle/utils/CommonUtilTest.java

```diff
@@ -212,14 +218,13 @@ public void testGetExistingConstructor() throws NoSuchMethodException {
 
     @Test
     public void testGetNonExistentConstructor() {
-        try {
-            CommonUtil.getConstructor(Math.class);
-            fail("IllegalStateException is expected");
-        }
-        catch (IllegalStateException expected) {
-            assertSame(NoSuchMethodException.class, expected.getCause().getClass(),
-                    "Invalid exception cause");
-        }
+        final IllegalStateException ex = assertThrows(IllegalStateException.class,
+                () -> CommonUtil.getConstructor(Math.class),
+                "IllegalStateException expected");
```

Comment:
I do not know why ... but I see that try-catch model is much easier to read.

Do you think new approach is better ?

Reply:
I do. `final Exception ex = assertThrows(...)` makes the tests unified:
```java
  foo = smth();
  assert(foo);
  ex = assertThrows();
  assert(ex);
```

Reply:
unified ? 
logical:
do/get Something
assert Something

not logical:
assertThrows
assert Exception

????

name of "assertThrow" is bad .....

better:
```
        final IllegalArgumentException ex = getThrowable(IllegalArgumentException.class,
                () -> CommonUtil.createPattern("[", Pattern.MULTILINE),
                "IllegalArgumentException expected");
        assertWithMessage("Invalid exception message")
                .that(ex)
                .hasMessageThat()
                        .isEqualTo("Failed to initialise regular expression [");
```

Reply:
I having hard to visually parse `() -> CommonUtil.createPattern("[", Pattern.MULTILINE),`
the main part for me as reader.

may be (imaginary code):
```
 final IllegalArgumentException ex = getThrowable(IllegalArgumentException.class)
     .at(() -> CommonUtil.createPattern("[", Pattern.MULTILINE))
```
would make it better.

or just:
```
final IllegalArgumentException ex = getThrowable(() -> CommonUtil.createPattern("[", Pattern.MULTILINE));
```

Reply:
I'm ok with this naming. Such method is not only `getThrowable`, it is more like to `getThrowableOrFailIfTheCodeDoesNotThrowAnException`, which is `assertThrows`.

The parameter "IllegalStateException expected" is really redundant, the default assertion message makes it clear which exception is expected. What about this:

```java
    public void testGetNonExistentConstructor() {
        final IllegalStateException ex = assertThrows(IllegalStateException.class, () -> {
            CommonUtil.getConstructor(Math.class);
        });

        assertWithMessage("Invalid exception cause")
                .that(ex)
                .hasCauseThat()
                        .isInstanceOf(NoSuchMethodException.class);
    }
```
Is this assertion more easy to read then the previous attempt?

Reply:
Yes, please use this in this PR.

Reply:
> Yes, please use this in this PR.

Done.